### ISSUE TITLE
duplexer_redux test does not fail meaningfully

### ISSUE THREAD

**Issue by Sequoia** [workshopper/stream-adventure#141]
```
$ stream-adventure verify duplexerredux.js

TAP version 13
# (anonymous)
ok 1 stream-adventure verify YOURFILE.js
#########################################
###   YOUR SOLUTION IS NOT CORRECT!   ###
#########################################


ok 2 solution exports a function
```

This is my file, currently (I've tried listening to the `end` event on every stream in this script, to no success):

``` javascript
var duplex = require('duplexer2');
var Writable = require('stream').Writable;

var totals = {};

var ws = Writable({objectMode:true});
ws._write = function (chunk, enc, next) {
  //console.log(chunk);
  if(totals[chunk.country]){
    totals[chunk.country]++;
  }else{
    totals[chunk.country] = 1;
  }
  //console.log(totals);
  next();
};

module.exports = function(counter){
  //console.log(counter);
  var plexer = duplex(ws, counter);
  plexer.on('end', function(){
    console.log('ending!!!');
    counter.setCounts(totals);
  });
  return plexer;
};
```
## the issue

The issue isn't my script or the logic error which I'm sure exists, the issue is **there's nothing to help me figure out what I'm doing wrong**. I can't even see what the third test is trying to test. :frowning:

**Comment by Sequoia** [comment]
I'm looking thru the `.verify` method and am still having trouble even isolating what my issue is.

**Comment by Sequoia** [comment]
figured out `plexer.finish`.

I think the solution here is "add a note that despite the name, `through2` can can be used to create *just writable* as well as duplex streams." I went to writable because I thought "well I don't want to put data **through**, I just want a writable stream.

**Comment by msfreemind** [comment]
I don't see where you're pulling the actual data in of the countries.

**Comment by howheels** [comment]
I was stuck on this problem for longer than I would like to admit for the same reason. It's also not very satisfying when you have the correct solution, because the output stream doesn't really expose the 'work' that's being done. I ended up creating my own test, building my own JSON stream based on the example data, and defining my own setCounts function like:
```
counter.setCounts = function (counts) {
    console.log(counts);
}
```

**Comment by kalinchernev** [comment]
Marking as suggestion as obviously the test framework needs an investigation how to give better feedback.

**Comment by ccarruitero** [comment]
fixed by #172
